internal/cmd/shell_bash.go

```diff
@@ -15,7 +15,7 @@ _direnv_hook() {
   trap - SIGINT;
   return $previous_exit_status;
 };
-if ! [[ "${PROMPT_COMMAND:-}" =~ _direnv_hook ]]; then
+if [[ ";${PROMPT_COMMAND-};" != *";_direnv_hook;"* ]]; then
```

Comment:
```suggestion
if [[ ";${PROMPT_COMMAND-};" != *"_direnv_hook"* ]]; then
```
The item might be alone in the list or at the start of the list.

Reply:
The added semicolons we have around the strings both on the left and right sides of the `!=` cover those cases, too.

If it's alone there, the test becomes `[[ ";_direnv_hook;" != *";_direnv_hook;"* ]]` which is false as wanted; if it's the first followed by something else it becomes `[[ ";_direnv_hook;something_else;" != *";_direnv_hook;"* ]]` which is also false as wanted.

Removing the semicolons per the suggestion would cause undesired results e.g. if `$PROMPT_COMMAND` would be `this_is_not_the_direnv_hook_we_are_looking_for`. Theoretical, but we could just do the right thing for that, too :)